### ISSUE TITLE
Request aria-label option to assist screen reader content

### ISSUE THREAD

**Issue by robeastwood-agency** [aehrc/smart-forms#1578]
**Is your feature request related to a problem? Please describe.**
The WCAG accessibility audit identified that it is highly desirable to have potentially different text for a given rendered question be configured for a screen reader. For example, the rendered text label for a question may be "_By when_" but the screen reader read out "_By when should the action be completed by?_". Another example would be a rendered display of a question answered count be "_2 / 3_" and the screen reader announce "_2 out of 3 questions answered in this section_".

**Describe the solution you'd like**
The accessibility advice is that the underlying html code may be something like this:
```html
<span 
    class="MuiTypography-root 
    MuiTypography-body1 css-p9lrf2" 
    id="tab-0" 
    aria-label="Section 1: Background">
    1. Background
</span>
```

And that 
> Given that the progress indicator component is reactive, likely need to bind the aria-label to the data source which might looks something like:
```html
<span 
    aria-label={`${answered} out of ${total} questions answered in this section`}>
  {answered} / {total}
</span>
```

To me, one of the challenges would be where the alternative screen reader text would be represented, by question. And superficially this would appear to be a candidate for a custom extension, given that an existing extension for this purpose doesn't look to be available (either in core FHIR spec nor SDC). So, a custom extension could be created that then can be added to specific question items where the alternate screen reader text is deemed necessary.

The other challenge would likely be that the renderer would need to honour that extension per question, such that the alternate text be added into the html `aria-label` attribute value.

All of this is somewhat 'imaginative', not knowing any better, so we would very much appreciate any views / advice / solutions etc. 

**Describe alternatives you've considered**
None

**Additional context**
I have searched SDC and zulip and found nothing close to this matter.

Thank you!

**Comment by fongsean** [comment]
Hi @robeastwood-agency, I think it makes sense to create a custom extension for this, given this just adds an "extra" aria-label that does not affect normal rendering behaviour on other SDC implementations.

This extension should be usable on all item labels, display items, and tab buttons.

Here's the structure I have on my mind:
```json
{
  "url": "https://smartforms.csiro.au/ig/StructureDefinition/QuestionnaireItemTextAriaLabelExpression",
  "valueExpression": Expression type
}
```
**This extension should be placed under `Questionnaire.item._text.extension`.**

Fixed strings - Tab buttons:
```json
{
  "url": "https://smartforms.csiro.au/ig/StructureDefinition/QuestionnaireItemTextAriaLabelExpression",
  "valueExpression": {
    "language": "text/fhirpath",
    "expression": "Section 1: Background"
  }
}
```

Dynamic strings - Progress indicator:
```json
{
  "url": "https://smartforms.csiro.au/ig/StructureDefinition/QuestionnaireItemTextAriaLabelExpression",
  "valueExpression": {
    "language": "text/fhirpath",
    "expression": "%BackgroundTabQCounter.toString() + ' out of 3 questions answered in this section'"
  }
}
```

Let me know what you think.

**Comment by robeastwood-agency** [comment]
Hi @fongsean - that solution looks perfect; manages static text as well as dynamic text - awesome!!

And presumably then the renderer needs to be updated to process the extension values - what are your thoughts there?

**Comment by robeastwood-agency** [comment]
Thanks @fongsean - we will shortly be updating to alpha.105 and will have the accessibility team look at it; we'll let you know

**Comment by robeastwood-agency** [comment]
@fongsean - testing has shown this works very well. This can be closed.
